### ISSUE TITLE
add support for webpack v4

### ISSUE THREAD

**Issue by mojoaxel** [soundcloud/chunk-manifest-webpack-plugin#55]
[Webpack v4 has been released](https://github.com/webpack/webpack/tags).
At the moment only webpack version 3 is supported:
* https://github.com/soundcloud/chunk-manifest-webpack-plugin/blob/master/package.json#L15

**Comment by IAMtheIAM** [comment]
Yes please, webpack 4 looks like it has many good features, we need chunk manifest plugin :-)

**Comment by rshackleton** [comment]
Experiencing the same issue as mentioned in https://github.com/webpack-contrib/mini-css-extract-plugin/issues/38, have narrowed the issue down to this plugin.

ERROR in chunk main [entry]
main-86b47c44b7412f8548be.css
Cannot read property 'chunkFilename' of undefined

**Comment by bf** [comment]
Any update on this issue? Waiting to use it with webpack 4.

**Comment by outshineamaze** [comment]
#59  we can use this, which support webpack 4.8.3 +

**Comment by fightingtheboss** [comment]
I ended up removing my dependency on this package in favour of using [`HtmlWebpackPlugin`](https://webpack.js.org/plugins/html-webpack-plugin/) with the following configs:

```javascript
// webpack.config.js
...
  plugins: [
    new HtmlWebpackPlugin({
      filename: 'webpack-common-manifest.json',
      template: 'manifest/template.html',
      inject: false,
    }),
  ],
```

```erb
<%# manifest/template.html %>
<%= JSON.stringify(webpack.assetsByChunkName) %>
```

This produced a JSON manifest referencing all chunks. You can use any of the other options for `HtmlWebpackPlugin` to select or exclude chunks, order, etc.

**Comment by 4lph4-Ph4un** [comment]
@fightingtheboss, is some part of that configuration missing? What makes webpack -object accessible from that template.html?

**Comment by fightingtheboss** [comment]
That variable is provided by `HtmlWebpackPlugin` while rendering the template. You can see documentation on that at the bottom of this section: https://github.com/jantimon/html-webpack-plugin#writing-your-own-templates

Turns out there's a much simpler solution to this problem though, simply use this plugin: https://github.com/danethurber/webpack-manifest-plugin. No need for a custom template in that case.

**Comment by 4lph4-Ph4un** [comment]
Ah! So it seems! I somehow missed that assuming it was solely for stats -object! Thanks for the trick! :) Unfortunately `webpack-manifest-plugin` is out of scope for me, since it has another compatibility issues currently related to my other configs (mainly `splitChunks: 'all'`).

**Comment by niemyjski** [comment]
Can we get one of the pr's for this merged in. Some look better than others... Can't even npm install after upgrading to npm4 or 5.
